### PR TITLE
PingManager improved to avoid unnecessary pings

### PR DESCRIPTION
**Pull request by MarcelHeckel**
Moved the "getLastStanzaReceived()" check inside the retry loop of the PingManager.pingServerIfNecessary() method. So unnecessary pings are avoided.

### CODE REVIEW

File: smack-extensions/src/main/java/org/jivesoftware/smackx/ping/PingManager.java [hunk 1]
```diff
@@ -419,21 +419,24 @@ public synchronized void pingServerIfNecessary() {
...
-            long now = System.currentTimeMillis();
-            // Delta since the last stanza was received
-            int deltaInSeconds = (int)  ((now - lastStanzaReceived) / 1000);
-            // If the delta is small then the ping interval, then we can defer the ping
-            if (deltaInSeconds < pingInterval) {
-                maybeSchedulePingServerTask(deltaInSeconds);
-                return;
-            }
-        }
+
```
Comment (vanitasvitae) [inline smack-extensions/src/main/java/org/jivesoftware/smackx/ping/PingManager.java]: This line should not contain any whitespace symbols.

File: smack-extensions/src/main/java/org/jivesoftware/smackx/ping/PingManager.java [hunk 2]
```diff
@@ -419,21 +419,24 @@ public synchronized void pingServerIfNecessary() {
...
-                maybeSchedulePingServerTask(deltaInSeconds);
-                return;
-            }
-        }
+       
         if (connection.isAuthenticated()) {
             boolean res = false;
 
             for (int i = 0; i < TRIES; i++) {
+            	{// check if there was an package meanwhile, then we don't need a ping...
```
Comment (vanitasvitae) [inline smack-extensions/src/main/java/org/jivesoftware/smackx/ping/PingManager.java]: That comment should probably go into the next line. Also there might be one too many parenthesis here?
Also that line is indented incorrectly.

**Comment by Flowdalic** [comment]
Thanks for your contribution. Please note that I have a private branch with changes which would conflict with this. Hence I'm putting this on hold until the private branch is merged (ETA end of the year).

**Comment by Flowdalic** [comment]
Looks good now. Could you squash your changes to a single commit please with a good commit message? As far as I can tell, with this change we assume that the connection is good even if there was no ping result but a stanzas within the ping interval. The commit message should clearly state that, ideally as short description in the commit title.

I am also curious about the background and motivation of this change. Did the current behavior ever cause problems for you?

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: c2ea78fde29147d06230c0a5dfcb9fd5ff677e05
  * Your commit message body contains a line that is longer than 72 characters
  * Your subject line is longer than 72 characters

Guidelines are available at https://github.com/igniterealtime/Smack/wiki/Guidelines-for-Smack-Developers-and-Contributors

---
This message was auto-generated by https://gitcop.com

**Comment by MarcelHeckel** [comment]
>  Could you squash your changes to a single commit please 

No, Git drives me crazy. I have made a new merge request because squashing was not possible due to conflicts: See https://github.com/igniterealtime/Smack/pull/295

> I am also curious about the background and motivation of this change. Did the current behavior ever cause problems for you?

In some case we have produced so much load, that the ping result was not delivered in time because so many other packets were delivered before.

**Comment by MarcelHeckel** [comment]
Sorry next try for merge request with a message of less the 72 characters.
https://github.com/igniterealtime/Smack/pull/296

**Comment by Flowdalic** [comment]
> Sorry next try for merge request with a message of less the 72 characters.

FYI You could have simply updated this PR by force pushing to its source branch.

**Pull request closed without merging**